substrafl/model_loading.py

```diff
@@ -147,93 +144,191 @@ def _get_train_task_from_round(client: substra.Client, compute_plan_key: str, ro
+        tag (str): Tag of the task to get.
 
-    return local_train_task
+    Returns:
+        substra.models.Task: The task matching the given requirements.
+    """
+    org_id = client.organization_info().organization_id
 
+    if rank_idx is not None:
+        filters = {"compute_plan_key": [compute_plan_key], "worker": [org_id], "rank": [rank_idx]}
+        local_tasks = client.list_task(filters=filters)
+        local_tagged_tasks = [t for t in local_tasks if t.tag == tag]
 
-def _load_algo(algo_path: Path, extraction_folder: Path) -> Any:
-    """Load into memory a serialized (and compressed (.tar.gz)) substrafl algo within the given algo_path.
+    else:
+        filters = {"compute_plan_key": [compute_plan_key], "worker": [org_id]}
+        local_tasks = client.list_task(filters=filters, order_by="end_date")
+        local_tagged_tasks = [t for t in local_tasks if t.tag == tag]
+
+    if len(local_tagged_tasks) == 0:
+        raise exceptions.TaskNotFoundError(
+            f"The given compute plan `{compute_plan_key}` has no {tag} task of rank {rank_idx} "
+            f"hosted on the organization {org_id}"
+        )
+
+    elif len(local_tagged_tasks) > 1:
+        raise exceptions.MultipleTaskError(
+            f"The given compute plan has {len(local_tagged_tasks)} {tag} tasks of rank {rank_idx}. "
+        )
+    local_task = local_tagged_tasks[0]
+
+    return local_task
+
+
+def _load_instance(gz_path: Path, extraction_folder: Path, remote: bool) -> Any:
+    """Load into memory a serialized (and compressed (.tar.gz)) SubstraFL Remote Struct instance within the
+    given path.
     This kind of file is usually the result of the ``substra.Client.download_function`` function applied to
-    a train task being part of a Substrafl experiment.
+    a task being part of a SubstraFL experiment.
 
     Args:
-        algo_path (Path): A file being the tar.gz compression of a substrafl RemoteStruct algorithm
+        gz_path (Path): A file being the tar.gz compression of a SubstraFL RemoteStruct instance
         extraction_folder (Path): Where to unpack the folder.
+        remote (bool): Whether to return the instance or the remote instance from the loaded remote_struct.
 
     Returns:
-        Any: The loaded substrafl object into memory.
+        Any: The loaded SubstraFL object into memory.
     """
-
-    with tarfile.open(algo_path, "r:gz") as tar:
+    with tarfile.open(gz_path, "r:gz") as tar:
         tar.extractall(path=extraction_folder)
 
     sys.path.append(str(extraction_folder))  # for local dependencies
 
     remote_struct = RemoteStruct.load(src=extraction_folder / SUBSTRAFL_FOLDER)
 
-    my_algo = remote_struct.get_instance()
+    if not remote:
+        instance = remote_struct.get_instance()
+    else:
+        instance = remote_struct.get_remote_instance()
+
+    return instance
+
 
-    return my_algo
+def _load_from_files(input_folder: os.PathLike, remote: bool = False) -> Any:
```

Comment:
why not use `pathlib.Path`? I have the impression it's the standard in substrafl